test/spec/smartbanner_spec.js

```diff
@@ -708,4 +730,34 @@ describe('SmartBanner', function() {
     });
   });
 
+  context('when api not set', function() {
+
+    before(function() {
+      global.window = new JSDOM(HTML, {userAgent: USER_AGENT_IPHONE_IOS9}).window;
+      global.document = window.document;
+      global.getComputedStyle = window.getComputedStyle;
+      smartbanner = new SmartBanner();
+    });
+
+    it('expected to smartbanner.apiEnabled is false ', function() {
+      expect(smartbanner.apiEnabled).to.be.false;
+    });
+
+  });
+
+  context('when api enabled', function() {
+
+    before(function() {
+      global.window = new JSDOM(HTML_API, {userAgent: USER_AGENT_IPHONE_IOS9}).window;
+      global.document = window.document;
+      global.getComputedStyle = window.getComputedStyle;
+      smartbanner = new SmartBanner();
+    });
+
+    it('expected to smartbanner.apiEnabled is true ', function() {
+      expect(smartbanner.apiEnabled).to.be.true;
```

Comment:
This test here is more about the options test that is already covered in option parser above. I'd replace it with a functional test and check if

- global object is added to `window` if API is turned on
- nothing is added to DOM if API is turned on
- component is added to DOM once `publish` is called 
- `smartbanner.view` event is only fired when API is used and `publish` gets called

Reply:
Unfortunately, I have no experience in writing functional tests.